Sources/EdgeNetworkHandlers/RequestBuilder.swift

```diff
@@ -80,13 +76,12 @@ class RequestBuilder {
         guard event.data != nil,
               let consents = event.data?[EdgeConstants.EventDataKeys.CONSENTS] as? [String: Any] else { return nil }
 
-        // set ECID if available
-        var identityMap = IdentityMap()
-        if let ecid = experienceCloudId {
-            identityMap.addItem(namespace: EdgeConstants.JsonKeys.ECID, id: ecid)
+        // set IdentityMap if available
+        var identityMap = [String: AnyCodable]()
+        if let identityMapDict = xdmPayloads.first(where: {$0.keys.first == EdgeConstants.SharedState.Identity.IDENTITY_MAP}) {
```

Comment:
This is a bit of an edge case, realistically we shouldn't know anything about the structure of the IdentityMap. We could do this by doing some custom encoding on the consent payload, but it seems a bit overkill given we aren't going to be attaching any additional XDM payloads on a consent request.

I think it's a fair compromise to say that Edge knows that name of "identityMap" so it can append to the Consent hit.

Reply:
I think it is correct to say we want to attach the identityMap to this request